soci/soci_index.go

```diff
@@ -265,16 +266,25 @@ func WithLegacyRegistrySupport(c *buildConfig) error {
 	return nil
 }
 
+// Speicifies the artifacts database
+func WithArtifactsDb(db *ArtifactsDb) BuildOption {
+	return func(c *buildConfig) error {
+		c.artifactsDb = db
+		return nil
+	}
+}
+
 // IndexBuilder creates soci indices.
 type IndexBuilder struct {
 	contentStore content.Store
 	blobStore    orascontent.Storage
+	ArtifactsDb  *ArtifactsDb
```

Comment:
do we need to export this field? (e.g. `ArtifactsDb` v.s. `artifactsDb`)

Reply:
i'm consuming it in a local change but I think I can try to use something else and un-export this. If that's possible i'll un-export this in the next pr.